File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char,int>mp;
        int res=0;
        int l=0,maxf=0;
        for(int i=0;i<s.size();i++){
            char c=s[i];
            mp[c]++;
            maxf=max(maxf,mp[c]);
            if((i-l+1)-maxf>k){
                mp[s[l]]--;
                l++;
            }
            res=max(res,i-l+1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-3.py
class Solution:
    def trap(self, height: List[int]) -> int:
        i=0
        n=len(height)-1
        l=height[i]
        r=height[n]
        area=0
        while i<n:
            if l<r:
                i+=1
                l=max(l,height[i])
                area+=l-height[i]
            else :
                n-=1
                r=max(r,height[n])
                area+=r-height[n]
        return area